uniqueids/admin.py

```diff
@@ -100,45 +139,61 @@ def save_model(self, request, obj, form, change):
             obj.valid = False
         else:
             for line in rows:
+                if ("address" in line and "address_type" in line):
+                    if (self.validate_address(line)):
+                        existing_address.add(line["address"])
+
                 missing_keys = self.validate_keys(line, obj.import_type)
 
                 if missing_keys:
                     for key in missing_keys:
                         missing_fields.add(key)
-                    obj.valid = False
+
+                missing_keys = self.validate_values(line, obj.import_type)
+
+                if missing_keys:
+                    for key in missing_keys:
+                        invalid_values.add(key)
```

Comment:
We seem to be overwriting `missing_keys` here, which is probably bad, and also a bit misleading as to what that variable now contains.

We can also simplify this to `invalid_values |= self.validate_values(line, obj.import_type)`